invenio_rdm_records/services/pids/providers/datacite.py

```diff
@@ -41,7 +61,18 @@ def __init__(self, name, client, pid_type="doi",
         )
 
         super(DataCiteProvider, self).__init__(
-            name, client, pid_type, default_status)
+            name, self.client, pid_type, default_status)
+
+    def _generate_id(self, recid, **kwargs):
```

Comment:
We need to pass, optionally, the record here. It was a request of a user who would like to customize the the suffix depending on some record fields. Can you add it?

Reply:
Wouldnt be enough to pass it as a kwargs? and then inherit and customize the `_generate_id` function?